core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -381,6 +382,36 @@ public Resolution resolve(AlluxioURI uri) throws InvalidPathException {
     }
   }
 
+  /**
+   * Reverse lookup given foreign URI and returns Alluxio path.
+   *
+   * @param foreignUri foreign URI
+   * @return Alluxio URI for given foreign path
+   * @throws InvalidPathException
+   */
+  public AlluxioURI translate(AlluxioURI foreignUri) throws InvalidPathException {
+    Preconditions.checkArgument(foreignUri.hasScheme(), "foreignUri.hasScheme()");
+
+    try (LockResource r = new LockResource(mReadLock)) {
+      LOG.debug("Translating foreign URI {}", foreignUri);
+      String uriStr = foreignUri.toString();
+      // Enumerate existing mount points to find a mount point that owns
+      // given uri.
+      for (Map.Entry<String, MountInfo> mountEntry : mState.getMountTable().entrySet()) {
+        AlluxioURI ufsUri = mountEntry.getValue().getUfsUri();
+        String ufsUriStr = ufsUri.toString();
+        // Check if current mount point owns given path.
+        if (uriStr.startsWith(ufsUriStr)) {
+          return new AlluxioURI(
+              PathUtils.concatPath(mountEntry.getKey(), uriStr.substring(ufsUriStr.length())));
+        }
+      }
+      // No mount found for given path.
+      throw new InvalidPathException(
+          String.format("Foreign URI: %s is not found on Alluxio mounts.", uriStr));
+    }
+  }
```

Comment:
Could you add tests for this method?

Reply:
https://github.com/Alluxio/alluxio/pull/9665/commits/ed9d1ca09901dfa3939ebdd1d0c2b71befbe8944